### DISCUSSION TITLE
Make_valid and substring functions

### DISCUSSION THREAD

**Discussion by DMuzer** [NetTopologySuite/NetTopologySuite#627] [Q&A] [answered]
Hi everyone! I try to find analogs of functions make_valid and substring of python shapely package in nts. Can anyone help me?

**Comment by FObermaier** [comment, marked as answer]
`make_valid` is `NetTopologySuite.Geometries.Utility.GeometryFixer.Fix(...)`. For `substring` have a look at the `NetTopologySuite.LinearReferencing` namespace.

**Reply by DMuzer** [reply]
Thank you!
